components/script/dom/window.rs

```diff
@@ -917,10 +918,15 @@ impl WindowMethods for Window {
 impl Window {
     pub fn get_runnable_wrapper(&self) -> RunnableWrapper {
         RunnableWrapper {
-            cancelled: Some(self.ignore_further_async_events.clone()),
+            cancelled: Some(self.ignore_further_async_events.borrow().clone()),
         }
     }
 
+    pub fn cancel_all_fetch_tasks(&self) {
+        let cancelled = mem::replace(&mut *self.ignore_further_async_events.borrow_mut(), Default::default());
+        cancelled.store(true, Ordering::Relaxed);
```

Comment:
This is cute. It's worth a comment explaining how this works (and mention that this cancels _all_ cancellable runnables related to this window, not just fetch tasks), because it was not immediately obvious to me.